### PR TITLE
Modernize Indium

### PR DESCRIPTION
**Pull request by IMS212**
FabricMC/fabric#2110 but for Indium.

### CODE REVIEW

File: src/main/java/link/infra/indium/renderer/render/BaseMeshConsumer.java [hunk 1]
```diff
@@ -82,44 +82,44 @@ public QuadEmitter getEmitter() {
 		return editorQuad;
 	}
 
-	private void renderQuad(MutableQuadViewImpl q) {
+	private void renderQuad(MutableQuadViewImpl quad) {
 		if (!transform.transform(editorQuad)) {
```
Comment (PepperCode1) [inline src/main/java/link/infra/indium/renderer/render/BaseMeshConsumer.java]:
```suggestion
		if (!transform.transform(quad)) {
```

File: src/main/java/link/infra/indium/mixin/renderer/MixinItemRenderer.java [hunk 2]
```diff
@@ -63,4 +63,4 @@ public void hook_method_23179(ItemStack stack, ModelTransformation.Mode transfor
 	public void fabric_renderBakedItemModel(BakedModel model, ItemStack stack, int light, int overlay, MatrixStack matrixStack, VertexConsumer buffer) {
 		renderBakedItemModel(model, stack, light, overlay, matrixStack, buffer);
 	}
-}
+}
```
Comment (PepperCode1) [inline src/main/java/link/infra/indium/mixin/renderer/MixinItemRenderer.java]: Need a newline here

File: src/main/java/link/infra/indium/renderer/render/BaseFallbackConsumer.java [hunk 3]
```diff
@@ -82,28 +80,17 @@ public void accept(BakedModel model) {
...
+		for (int i = 0; i <= ModelHelper.NULL_FACE_ID; i++) {
+			final Direction cullFace = ModelHelper.faceFromIndex(i);
+			final List<BakedQuad> quads = model.getQuads(blockState, cullFace, random.get());
 			final int count = quads.size();
 
 			if (count != 0) {
-				for (int j = 0; j < count; j++) {
-					final BakedQuad q = quads.get(j);
-					renderQuad(q, face, defaultMaterial);
+				for (final BakedQuad q : quads) {
```
Comment (PepperCode1) [inline src/main/java/link/infra/indium/renderer/render/BaseFallbackConsumer.java]: Revert the change to an enhanced for loop; this is inconsistent with Indigo and less efficient.

File: src/main/java/link/infra/indium/renderer/render/ItemRenderContext.java [hunk 4]
```diff
@@ -251,4 +317,4 @@ public QuadEmitter getEmitter() {
 		editorQuad.clear();
 		return editorQuad;
 	}
-}
+}
```
Comment (PepperCode1) [inline src/main/java/link/infra/indium/renderer/render/ItemRenderContext.java]: Need a newline here

**Review by PepperCode1 — CHANGES_REQUESTED** [review]

File: src/main/java/link/infra/indium/renderer/render/BaseMeshConsumer.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/link/infra/indium/renderer/render/BaseMeshConsumer.java]: Are you sure this is right?

**Review by PepperCode1 — APPROVED** [review]

**Comment by comp500** [comment]
```java
 * Does not implement emissive lighting for sake
 * of simplicity in the default renderer.
```
This javadoc note in `ItemRenderContext` was removed in the original Indigo PR; does it still apply to Indium with these changes or should it be removed here as well?
Edit: I'm guessing it should be removed as `renderQuadEmissive` exists :P

**Pull request merged**